pkg/watcher/webhook_chart.go

```diff
@@ -53,31 +56,27 @@ func resolveKcpAddr(kcpConfig *rest.Config, managerConfig *SkrWebhookManagerConf
 		return net.JoinHostPort(defaultK3dLocalhostMapping, strconv.Itoa(managerConfig.LocalGatewayHTTPPortMapping)),
 			nil
 	}
-	// Get public KCP IP from the ISTIO load balancer external IP
-	kcpClient, err := client.New(kcpConfig, client.Options{})
+	ctx := context.TODO()
+
+	// Get public KCP DNS name and port from the Gateway
+	kcpIstioClient, err := versionedclient.NewForConfig(kcpConfig)
 	if err != nil {
 		return "", err
 	}
-	ctx := context.TODO()
-	loadBalancerService := &corev1.Service{}
-	if err := kcpClient.Get(ctx, client.ObjectKey{
-		Name:      managerConfig.IstioIngressServiceName,
-		Namespace: managerConfig.IstioNamespace,
-	}, loadBalancerService); err != nil {
+
+	gateway, err := kcpIstioClient.NetworkingV1beta1().Gateways(managerConfig.IstioGatewayNamespace).
```

Comment:
I'm not recommend to import `kcpIstioClient`, instead, what you can do is, add istio schema in main.go `init()`, then you can use `kcpClient` to get this istio gateway.

```
	utilruntime.Must(istiov1beta1.AddToScheme(scheme))
```